packages/xod-project/src/flatten.js

```diff
@@ -158,7 +158,7 @@ export const extractLeafPatches = def(
   (project, path, patch) =>
     R.cond([
       [
-        isLeafPatchWithImplsOrTerminal,
+        R.either(isLeafPatchWithImplsOrTerminal, Patch.isAbstractPatch),
```

Comment:
This one looks like it could break the Project, but I already now that flattening function is called twice, and `isAbstractPath` could be true at the first run, and then there will be no abstract patches (only specialization patches will be left) and they will be processed as usual at the second run. But I know it cause you tell me about it yesterday :)
How about to leave a comment?